tardis/montecarlo/src/cmontecarlo.c

```diff
@@ -364,8 +372,35 @@ increment_j_blue_estimator (const rpacket_t * packet, storage_model_t * storage,
 #ifdef WITHOPENMP
 #pragma omp atomic
 #endif
+
   storage->line_lists_j_blues[j_blue_idx] +=
-    comov_energy / rpacket_get_nu (packet);
+  comov_energy / rpacket_get_nu (packet);
+  double key = rpacket_get_nu (packet) * doppler_factor;
+  double *p2 = (double *) bsearch(&key,storage->line_list_nu , storage->no_of_lines, sizeof(storage->line_list_nu[0]), compare);
+  if (p2){
+      count_resonance++;
```

Comment:
@SooryaN - the counter now resides at the most logical location in the program flow. However, you are making the counting task unnecessarily complicated. See comment to line 376!